src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,234 @@
+package duke;
+
+import duke.exceptions.IncorrectFormatException;
+import duke.exceptions.UnrecognizedCommandException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+import java.util.Scanner;
+
+public class Duke {
+
+    static TaskManager taskManager = new TaskManager();
+    static Scanner scanner = new Scanner(System.in);
+    static FileDataManager fileDataManager = new FileDataManager();
+
+    public static void main(String[] args) {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+
+        fileDataManager.loadData(taskManager);
+        greet();
+        processInput();
+    }
+
+    public static void printLine() {
+        System.out.print("\t");
+        for (int i = 0; i < 60; i++) {
+            System.out.print("\u2500");
+        }
+        System.out.println();
+    }
+
+    public static void greet() {
+        printLine();
+        System.out.println("\tHello! I'm Duke!");
+        System.out.println("\tWhat can I do for you?");
+        printLine();
+    }
+
+    public static void processInput() {
+        String input = acceptAndValidateInput();
+        String command = retrieveCommand(input);
+        String parameters = retrieveParameters(input);
+        boolean validFormat;
+
+        while (!command.equals("bye")) {
+            printLine();
+            validFormat = false;
+            try {
+                validateFormat(command, parameters);
+                validFormat = true;
+            } catch (NumberFormatException e) {
+                System.out.println("\tIncorrect format! Please enter a valid integer number to mark a task as done or not done!");
+            } catch (IncorrectFormatException e) {
+                System.out.println("\t" + e.getMessage());
+            }
+            if (validFormat) {
+                executeCommand(command, parameters);
+            }
+            printLine();
+            input = acceptAndValidateInput();
+            command = retrieveCommand(input);
+            parameters = retrieveParameters(input);
+        }
+        exit(0);
+    }
+
+    public static void executeCommand(String command, String parameters) {
```

Comment:
This function is very long (>30 lines). Maybe break it up into multiple, smaller functions